packages/components/src/components/stepper/stepper.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, Prop, Element, State, EventEmitter, Event} from "@stencil/core";
+
+@Component({
+    tag: 'ifx-stepper',
+    styleUrl: 'stepper.scss',
+    shadow: true
+})
+
+export class Stepper{
+
+    @Prop() showNumber: boolean = false;
+    @Prop() activeStep: number = 1;
+    @State() internalActiveStep = undefined;
+    @Prop() variant: string = 'default';
+    @Element() el: HTMLElement;
+    @Event() ifxActiveStepChange: EventEmitter;
+    private stepsCount: number;
+
+
+    // Syncing children (steps) with parent state
+    updateChildren(){
+        const steps: NodeListOf<HTMLIfxStepElement> = this.el.querySelectorAll('ifx-step');
+        for(let i = 0; i < steps.length; i++){
+            steps[i].stepperState = {activeStep: this.internalActiveStep, showNumber: this.showNumber, variant: this.variant};
+        }
+        
+    }
+
+    addStepIdsToStepsAndCountSteps(){
+        const steps: NodeListOf<HTMLIfxStepElement> = this.el.querySelectorAll('ifx-step');
+        steps[steps.length-1].lastStep = true;
+        for(let i = 0; i < steps.length; i++){
+            steps[i].stepId = i+1;
+        }
+        this.stepsCount = steps.length;
+    }
+
+    updateActiveStep(){
+        let newActiveStep = Math.max(1, Math.min(this.stepsCount+(this.variant !== 'compact' ? 1 : 0), this.activeStep));
+
+        if(this.internalActiveStep && newActiveStep != this.internalActiveStep){
+            this.internalActiveStep = newActiveStep;
+            this.ifxActiveStepChange.emit({activeStep: this.internalActiveStep, totalSteps: this.stepsCount});
+        }else{
+            this.internalActiveStep = newActiveStep;
+        }
+    }
+    
+    componentWillLoad(){
+        this.addStepIdsToStepsAndCountSteps();
+        this.updateActiveStep();
+        this.updateChildren();
+    }
+    
+    componentWillUpdate(){
+        this.updateActiveStep();
```

Comment:
From an efficiency perspective, it might be more beneficial to use Stencil's @Watch decorator to observe changes in specific props rather than calling the updateActiveStep() method in the lifecycle methods. The componentWillUpdate() and componentWillLoad() will be called whenever any state or prop changes, not just when activeStep changes, potentially leading to unnecessary calls to updateActiveStep.
Using @Watch, you only execute a method whenever a specific prop changes. This way, you can ensure that the updateActiveStep() method is only called when the activeStep prop changes.

```
@Prop() activeStep: number = 1;
@Watch('activeStep')
activeStepHandler(newValue: number, oldValue: number) {
    // This will be executed whenever the `activeStep` prop changes.
    this.updateActiveStep();
}
```